HighwayPatrol.cpp

```diff
@@ -1,16 +1,20 @@
 #include "HighwayPatrol.h"
 #include <iostream>
+#include "Car.h"
+#include "Motorcycle.h"
+#include "SemiTruck.h"
 
-HighwayPatrol::HighwayPatrol() : Vehicle("HighwayPatrol")
-{
+HighwayPatrol::HighwayPatrol() : Vehicle("HighwayPatrol") { }
 
-}
+HighwayPatrol::~HighwayPatrol() = default;
+HighwayPatrol::HighwayPatrol (const HighwayPatrol&) = default;
+HighwayPatrol& HighwayPatrol::operator = (const HighwayPatrol&) = default;  
 
 void HighwayPatrol::scanHighway(Highway* h)
 {
     std::cout << name << ": scanning highway for speeders" << std::endl;
 
-    for( size_t i = h->vehicles.size(); --i >= 0; )
+    for( size_t i = h->vehicles.size() - 1; static_cast<int>(i) >= 0; --i)
```

Comment:
While this works, what I had after fixing the bug another person spotted here (which didn't get pushed to your project) is this:
```
for( size_t i = h->vehicles.size(); i-- > 0; )
```
Take a look here: http://coliru.stacked-crooked.com/a/1c450220662e32f8 
- [ ] revise to eliminate that `static_cast<>`

Reply:
Cool I had tried something similar but figured you would not want i--  haha.